### PR TITLE
Parameterize sidecar image name for Kubernetes scripts.

### PR DESCRIPTION
**Pull request by davidbtucker**

### CODE REVIEW

File: kube/patch.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ spec:
     spec:
       containers:
       - name: sidecar
-        image: gcr.io/stackdriver-prometheus/stackdriver-prometheus-sidecar:${SIDECAR_IMAGE_TAG}
```
Comment (jkohen) [inline kube/patch.sh]: This script is used in the public docs. We shouldn't change the API. Can you preserve the current value as a default value?

File: kube/full/deploy.sh [hunk 2]
```diff
@@ -5,7 +5,8 @@ set -u
 
 pushd "$(dirname "$0")"
 
-# Override to use a different Docker image version for the sidecar.
+# Override to use a different Docker image name or version for the sidecar.
+export SIDECAR_IMAGE_NAME=${SIDECAR_IMAGE_NAME:-'gcr.io/stackdriver-kubernetes-1337/stackdriver-prometheus-sidecar'}
```
Comment (jkohen) [inline kube/full/deploy.sh]: This is changing the default value in kube/patch.sh. Can you justify the change?

File: kube/patch.sh (hunk 1 above)
Comment (davidbtucker) [reply kube/patch.sh]: Sure, moved it over to patch.sh.

File: kube/full/deploy.sh (hunk 2 above)
Comment (davidbtucker) [reply kube/full/deploy.sh]: I moved var over to patch.sh, so it should be backwards compatible now.

**Comment by davidbtucker** [comment]
> It's still different, it should be `gcr.io/stackdriver-prometheus/stackdriver-prometheus-sidecar`, right? Can you please test the script so we are sure it still works? This is part of the public documentation and how most users try our Stackdriver integration first, so it is important that it works.

I think you were looking at an old snapshot.

In any case, I've tested the latest version and it seems to work.

**Pull request merged**